src/ape/contracts/base.py

```diff
@@ -12,29 +14,29 @@
 )
 from ape.logging import logger
 from ape.types import AddressType
-from ape.utils import dataclass
+from ape.utils import AbstractBaseModel
 
 if TYPE_CHECKING:
     from ape.managers.converters import ConversionManager
     from ape.managers.networks import NetworkManager
 
 
-@dataclass
-class ContractConstructor:
+class ContractConstructor(AbstractBaseModel):
     deployment_bytecode: bytes
     abi: ConstructorABI
     provider: ProviderAPI
-    converter: "ConversionManager"
 
-    def __post_init__(self):
-        if len(self.deployment_bytecode) == 0:
+    @validator("deployment_bytecode")
+    def check_empty_contract(cls, value):
+        if len(value) == 0:
             logger.warning("Deploying an empty contract (no bytecode)")
+        return value
```

Comment:
I updated the logic, but I'm not so sure about the `Optional[bytes]` because it causes a trickle effect on the ecosystem/network that we might not want.